.eslintrc.yml

```diff
@@ -13,8 +13,11 @@ plugins:
   - '@typescript-eslint'
   - 'jsdoc'
 rules:
-    semi: "error"
-    no-extra-semi: "warn"
-    curly: "warn"
-    quotes: ["error", "single", { "allowTemplateLiterals": true } ]
-    eqeqeq: "error"
+  semi: "error"
+  no-extra-semi: "warn"
+  curly: "warn"
+  quotes: ["error", "single", { "allowTemplateLiterals": true } ]
+  eqeqeq: "error"
+  no-empty: "off"
```

Comment:
Ignoring these rules can be potential bugs.
The empty block should be implemented before merging pulls.
`no-inferrable-types` can be defined by the `type` or `interface`.
no-unused-vars will lots the code readability

Reply:
I agree about the `no-unused-vars`, the other two rules I disabled can be a bit annoying though:
* `no-inferrable-types`: The point of this rule is to reduce unnecessary verbosity. Verbosity could make code harder to read, but so can missing (explicit) type definitions, so I think this should be left to the programmer to decide.
* `no-empty`: The only instance where I encounter this rule are empty `catch` blocks and empty callbacks passed to functions (when these cannot be left out). The rule can be circumvented by adding a comment in the otherwise empty block, but that adds e.g. two extra lines to an empty catch block and still allows people to leave empty functions in the code.

Reply:
Yes, your examples point to the limitation of the current rule. So how is using rule options?
In this case, we can be staying in the TypeScript standard.

* For the `no-inferrable-types`, we can use  `ignoreProperties` option. I think `ignoreParameters` option is also good in the future.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-inferrable-types.md#ignoreparameters
* For the `no-empty`, then we can use the `allowEmptyCatch` option. https://eslint.org/docs/rules/no-empty#allowemptycatch

Reply:
These rules pretty much cover all my annoyances, thanks for pointing them out :)